design-proposals/kubebmc.md

```diff
@@ -0,0 +1,156 @@
+# Overview
+
+This proposal introduces [KubeBMC](https://github.com/starbops/kubebmc), a BMC emulator for Kubernetes that facilitates the deployment of software/applications/platforms such as OpenShift and OpenStack - whose installers typically require communication with bare-metal out-of-band management protocols like IPMI and Redfish - in KubeVirt VMs for development, testing, and debugging purposes, similar to the functionality provided by [VirtualBMC](https://opendev.org/openstack/virtualbmc) and [sushy-emulator](https://opendev.org/openstack/sushy-tools) but within a Kubernetes context.
+
+## Motivation
+
+Developers require the ability to deploy OpenShift and OpenStack clusters on local virtual environments. Existing solutions involve libvirt domains or QEMU VMs with BMC emulators, which are not directly compatible with KubeVirt due to its internal use of [`virtqemud`](https://libvirt.org/manpages/virtqemud.html), necessitating a Kubernetes-native solution. This was initially raised in the GitHub issue: [Expose a virtual BMC service per virtual machine, for IPMI and/or Redfish actions · Issue #9108 · kubevirt/kubevirt](https://github.com/kubevirt/kubevirt/issues/9108).
```

Comment:
Is it the only use case?

Reply:
AFAIK relevant projects like Virtual BMC and sushy-tools were created to test bare-metal provision-related tools/libraries. I would say it's the primary use case.

Reply:
Just thinking, if we could add some more. But it's fine for now!

Reply:
Maybe we can generalize it a bit (e.g. we want the ability so simulate baremetal environments in general, not just OpenStack and OpenShift) or add some more examples. I saw Metal3 mentioned already, it could be added here also. We would like to use it to simulate baremetal kubernetes cluster through CAPI/CAPM3.

Reply:
I forgot to mention Metal3 here :doh:
Yeah I think in this paragraph it's better to have a general description.